environment/config.go

```diff
@@ -39,18 +38,44 @@ import (
 	"github.com/m3db/m3x/instrument"
 )
 
+const (
+	defaultSDTimeout = 10 * time.Minute
+)
+
 var (
 	errNilRetention  = errors.New("namespace retention options cannot be empty")
 	errMissingConfig = errors.New("must supply service or static config")
 )
 
 // Configuration is a configuration that can be used to create namespaces, a topology, and kv store
 type Configuration struct {
-	// Service is used when a topology initializer is not supplied.
-	Service *etcdclient.Configuration `yaml:"service"`
-
 	// StaticConfiguration is used for running M3DB with a static config
 	Static *StaticConfiguration `yaml:"static"`
+
+	// KV sets the kv configuration for either a clustered or embedded etcd
+	KV *KVConfig `yaml:"kv"`
+}
+
+// KVConfig sets the configs for the KV client and server
+type KVConfig struct {
+	Client *etcdclient.Configuration `yaml:"client"`
+
+	// Presence of a (etcd) server in this config denotes an embedded cluster
+	Server *EmbeddedKV `yaml:"server"`
```

Comment:
Lets keep server/client configs in different structs: leave `Service` where it was; and add EmbeddedKV to the top-level `Configuration` struct.